lib/puppet/functions/stdlib/ip_in_range.rb

```diff
@@ -0,0 +1,29 @@
+# Returns true if the ipaddress is within the given CIDRs
+#
+# @example ip_in_range(<IPv4 Address>, <IPv4 CIDR>)
+#   stdlib::ip_in_range('10.10.10.53', '10.10.10.0/24') => true
+Puppet::Functions.create_function(:'stdlib::ip_in_range') do
+  # @param [String] ipaddress Get an IP address
+  # @param [Variant[String, Array]] range Get a one or many ranges of IP addresses
+  #
+  # @return [Boolean] True or False
+  dispatch :ip_in_range do
+    param 'String', :ipaddress
+    param 'Variant[String, Array]', :range
+    return_type 'Boolean'
+  end
+
+  def ip_in_range(ipaddress, range)
+    require 'ipaddr'
```

Comment:
calling `require` every time the function is called is unnecessary - better to call this in the main body - for example move this to line 15. That way it is only called once.

Reply:
got it!